build/lib/src/builder/build_step_impl.dart

```diff
@@ -85,15 +85,27 @@ class BuildStepImpl implements BuildStep {
   @override
   Future writeAsBytes(AssetId id, FutureOr<List<int>> bytes) {
     _checkOutput(id);
-    var done = _writer.writeAsBytes(id, bytes);
+    Future done;
```

Comment:
You **may** want to encapsulate this FutureOr handling logic in a function, but its only copied twice for now so up to you.